kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/AdminResource.scala

```diff
@@ -72,6 +73,67 @@ private[v1] class AdminResource extends ApiRequestContext with Logging {
       @QueryParam("sharelevel") shareLevel: String,
       @QueryParam("subdomain") subdomain: String,
       @QueryParam("hive.server2.proxy.user") hs2ProxyUser: String): Response = {
+    val userName = getUserName(fe, hs2ProxyUser)
+    val engineSpace =
+      getEngineSpace(userName, engineType, shareLevel, subdomain, "default", fe.getConf)
+
+    withDiscoveryClient(fe.getConf) { discoveryClient =>
+      val engineNodes = discoveryClient.getChildren(engineSpace)
+      engineNodes.foreach { node =>
+        val nodePath = s"$engineSpace/$node"
+        info(s"Deleting engine node:$nodePath")
+        try {
+          discoveryClient.delete(nodePath)
+        } catch {
+          case e: Exception =>
+            error(s"Failed to delete engine node:$nodePath", e)
+            throw new NotFoundException(s"Failed to delete engine node:$nodePath," +
+              s"${e.getMessage}")
+        }
+      }
+    }
+
+    Response.ok(s"Engine $engineSpace is deleted successfully.").build()
+  }
+
+  @ApiResponse(
+    responseCode = "200",
+    content = Array(new Content(
+      mediaType = MediaType.APPLICATION_JSON)),
+    description = "delete kyuubi engine")
+  @GET
+  @Path("engine")
+  def listEngine(
+      @QueryParam("type") engineType: String,
+      @QueryParam("sharelevel") shareLevel: String,
+      @QueryParam("subdomain") subdomain: String,
+      @QueryParam("hive.server2.proxy.user") hs2ProxyUser: String): Seq[ServiceNodeInfo] = {
```

Comment:
Please define one class named `ServiceNodeInfo` in `org.apache.kyuubi.client.api.v1.dto` package.

Reply:
in kyuubi-rest-client module.

Reply:
The fields for class `Engine`.
- version
- engineType
- sharelevel
- subdomain
- instance

how do you think about? @pan3793

Reply:
<img width="1528" alt="image" src="https://user-images.githubusercontent.com/6757692/196105045-bc7bcb55-edb6-4024-a88b-bbd5af6f4803.png">
For the refInstanceId, It looks meaningless for me @pan3793